quartz/src/main/java/com/vitorpamplona/quartz/utils/LargeCache.kt

```diff
@@ -76,489 +76,7 @@ class LargeCache<K, V> {
-        return runner.results
-    }
-
-    fun <R> mapNotNullIntoSet(consumer: BiMapper<K, V, R?>): Set<R> {
-        val runner = BiMapUniqueCollector(consumer)
-        innerForEach(runner)
-        return runner.results
-    }
-
-    fun <R> mapFlatten(consumer: BiMapper<K, V, Collection<R>?>): List<R> {
-        val runner = BiMapFlattenCollector(consumer)
-        innerForEach(runner)
-        return runner.results
-    }
-
-    fun <R> mapFlattenIntoSet(consumer: BiMapper<K, V, Collection<R>?>): Set<R> {
-        val runner = BiMapFlattenUniqueCollector(consumer)
-        innerForEach(runner)
-        return runner.results
-    }
-
-    fun maxOrNullOf(
-        filter: BiFilter<K, V>,
-        comparator: Comparator<V>,
-    ): V? {
-        val runner = BiMaxOfCollector(filter, comparator)
-        innerForEach(runner)
-        return runner.maxV
-    }
-
-    fun sumOf(consumer: BiSumOf<K, V>): Int {
-        val runner = BiSumOfCollector(consumer)
-        innerForEach(runner)
-        return runner.sum
-    }
-
-    fun sumOfLong(consumer: BiSumOfLong<K, V>): Long {
-        val runner = BiSumOfLongCollector(consumer)
-        innerForEach(runner)
-        return runner.sum
-    }
-
-    fun <R> groupBy(consumer: BiNotNullMapper<K, V, R>): Map<R, List<V>> {
-        val runner = BiGroupByCollector(consumer)
-        innerForEach(runner)
-        return runner.results
-    }
-
-    fun <R> countByGroup(consumer: BiNotNullMapper<K, V, R>): Map<R, Int> {
-        val runner = BiCountByGroupCollector(consumer)
-        innerForEach(runner)
-        return runner.results
-    }
-
-    fun <R> sumByGroup(
-        groupMap: BiNotNullMapper<K, V, R>,
-        sumOf: BiNotNullMapper<K, V, Long>,
-    ): Map<R, Long> {
-        val runner = BiSumByGroupCollector(groupMap, sumOf)
-        innerForEach(runner)
-        return runner.results
-    }
-
-    fun count(consumer: BiFilter<K, V>): Int {
-        val runner = BiCountIfCollector(consumer)
-        innerForEach(runner)
-        return runner.count
-    }
-
-    fun <T, U> associate(transform: (K, V) -> Pair<T, U>): Map<T, U> {
-        val runner = BiAssociateCollector(size(), transform)
-        innerForEach(runner)
-        return runner.results
-    }
-
-    fun <T, U> associateNotNull(transform: (K, V) -> Pair<T, U>?): Map<T, U> {
-        val runner = BiAssociateNotNullCollector(size(), transform)
-        innerForEach(runner)
-        return runner.results
-    }
-
-    fun <U> associateWith(transform: (K, V) -> U?): Map<K, U?> {
-        val runner = BiAssociateWithCollector(size(), transform)
-        innerForEach(runner)
-        return runner.results
-    }
-
-    fun <U> associateNotNullWith(transform: (K, V) -> U): Map<K, U> {
-        val runner = BiAssociateNotNullWithCollector(size(), transform)
-        innerForEach(runner)
-        return runner.results
-    }
-
-    private fun innerForEach(runner: BiConsumer<K, V>) {
-        cache.forEach(runner)
```

Comment:
Just make sure this forEach is the one from ConcurrentSkipListMap and not from the other kotlin wrappers/extensions. The original is almost 2 times faster than the extension functions